File: app/src/main/java/io/github/marcocipriani01/telescopetouch/activities/SplashScreenComponent.java
package io.github.marcocipriani01.telescopetouch.activities;

import dagger.Component;
import io.github.marcocipriani01.telescopetouch.ApplicationComponent;
import io.github.marcocipriani01.telescopetouch.activities.dialogs.EulaDialogFragment;
import io.github.marcocipriani01.telescopetouch.activities.dialogs.WhatsNewDialogFragment;
import io.github.marcocipriani01.telescopetouch.inject.PerActivity;

/**
 * Created by johntaylor on 4/2/16.
 */
@PerActivity
@Component(modules = SplashScreenModule.class, dependencies = ApplicationComponent.class)
public interface SplashScreenComponent extends EulaDialogFragment.ActivityComponent,
        WhatsNewDialogFragment.ActivityComponent {
    void inject(SplashScreenActivity activity);
}

File: app/src/main/java/com/google/android/stardroid/activities/SplashScreenComponent.kt
package com.google.android.stardroid.activities

import com.google.android.stardroid.ApplicationComponent
import com.google.android.stardroid.activities.dialogs.EulaDialogFragment
import com.google.android.stardroid.activities.dialogs.WhatsNewDialogFragment
import com.google.android.stardroid.inject.PerActivity
import dagger.Component

/**
 * Created by johntaylor on 4/2/16.
 */
@PerActivity
@Component(modules = [SplashScreenModule::class], dependencies = [ApplicationComponent::class])
interface SplashScreenComponent : EulaDialogFragment.ActivityComponent,
    WhatsNewDialogFragment.ActivityComponent {
    fun inject(activity: SplashScreenActivity?)
}